### PR TITLE
Fixing precompiled function discovery bug

### PR DESCRIPTION
**Pull request by mathewc**
Fix for https://github.com/Azure/azure-webjobs-sdk/issues/1525

Related to https://github.com/Azure/azure-functions-host/issues/911. These changes will allow instance method functions for _precompiled_ functions, but for csx compiled functions we still don't support it (until #911) is fixed, and we change our IL gen to instantiate instances. However I don't think we should block on that in supporting this for precompiled functions now, since it opens the door to lots of SDK scenarios users expect to work (like function invocation filters).

### CODE REVIEW

File: src/WebJobs.Script/Description/DotNet/Compilation/Raw/RawAssemblyCompilation.cs [hunk 1]
```diff
@@ -53,7 +53,7 @@ public FunctionSignature GetEntryPointSignature(IFunctionEntryPointResolver entr
                 throw new InvalidOperationException($"The function type name '{typeName}' is invalid.");
             }
 
-            MethodInfo method = functionType.GetMethod(methodName, BindingFlags.Static | BindingFlags.Public);
+            MethodInfo method = functionType.GetMethod(methodName, BindingFlags.Public | BindingFlags.Static | BindingFlags.Instance | BindingFlags.DeclaredOnly);
```
Comment (mathewc) [inline src/WebJobs.Script/Description/DotNet/Compilation/Raw/RawAssemblyCompilation.cs]: Using the same flags that the core SDK uses (https://github.com/Azure/azure-webjobs-sdk/blob/v2.x/src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs#L27)

Comment (fabiocav) [reply src/WebJobs.Script/Description/DotNet/Compilation/Raw/RawAssemblyCompilation.cs]: This doesn't just apply to direct load, but to the raw assembly model as well. We need to ensure we have a test that validates that those classes will be properly instantiated to call instance methods. Did you have a chance to test that?

**Comment by paulbatum** [comment]
@mathewc @fabiocav whats the status of this PR?

**Comment by mathewc** [comment]
I need to try the raw assembly scenario as Fabio suggested, and ensure it works or get it to work. Shouldn't be that big of an issue - might need a minor IL gen tweak.

**Comment by fabiocav** [comment]
Closing this PR as it is stale. Some of this work will be done in the context of the DI work happening.

**Pull request closed without merging**